### PR TITLE
Fixes crash  :zap:

### PR DESCRIPTION
**Pull request by millerscout**
fixes: 

https://mods.factorio.com/mod/Rampant/discussion/64fd9c2f6ca8358ef4a69f70
https://mods.factorio.com/mod/Rampant/discussion/65071df332462785b74e7feb
https://mods.factorio.com/mod/Rampant/discussion/6502e138b9d51e3e1baa4598
https://mods.factorio.com/mod/Rampant/discussion/64fcfe1055fe6cfab4001c47

### CODE REVIEW

**Comment by veden** [comment]
first, thank you for taking the time for making this bugfix.

File: control.lua [hunk 1]
```diff
@@ -965,7 +965,7 @@ end
 
 -- hooks
 
-script.on_event(defines.events.on_tick,
+script.on_nth_tick(30,
```
Comment (veden) [inline control.lua:968]: Doesn't this change break the logic inside the function?
tick now only falls into multiples of 30 and the logic uses all ticks

Comment (millerscout) [reply control.lua:968]: I'm the one who thank you!

i'll revert the change about the on_nth_tick for now.

i've played for 2-3 hours and i haven't noticed anything awful yet.

Comment (millerscout) [reply control.lua]: I rollback the event trigger @veden 

I will continue to further investigate and profile before hand 😉

Comment (PasekThePolak) [reply control.lua]: My game was crashing due to this error. made the changes to line 494 and left 968 as `script.on_event(defines.events.on_tick,`

and my game still crashed.

Comment (millerscout) [reply control.lua]: hello!
that line doesn't fix the problem, that's why

https://github.com/veden/Rampant/pull/178/files#diff-12b76a5b1d63b6a1a18620c4fd67f98aa90ae107b0265c6f772614b87b540af0L494 << you need to change like that

**Comment by Frag31337** [comment]
Hm, I think that 518 line need this fix too.
if cause and cause.valid and cause.get_health_ratio() ~= nil then

**Comment by millerscout** [comment]
> Hm, I think that 518 line need this fix too. if cause and cause.valid and cause.get_health_ratio() ~= nil then

I'll check asap, thanks!

**Comment by Kirill** [comment]
https://mods.factorio.com/mod/Rampant/discussion/650f085a08fc2c2aa7464855
and https://mods.factorio.com/mod/Rampant/discussion/650ded2eabdb2d1e559eef94 (approve this fixed)

**Pull request merged**